src/test/kotlin/no/nav/syfo/identhendelse/IdenthendelseServiceSpek.kt

```diff
@@ -0,0 +1,145 @@
+import no.nav.syfo.client.pdl.PdlClient
+import no.nav.syfo.dialogmotekandidat.database.createDialogmotekandidatStoppunkt
+import no.nav.syfo.dialogmotestatusendring.database.createDialogmoteStatus
+import no.nav.syfo.dialogmotestatusendring.domain.DialogmoteStatusEndring
+import no.nav.syfo.dialogmotestatusendring.domain.DialogmoteStatusEndringType
+import no.nav.syfo.domain.PersonIdentNumber
+import no.nav.syfo.identhendelse.database.getIdentOccurenceCount
+import no.nav.syfo.testhelper.ExternalMockEnvironment
+import no.nav.syfo.testhelper.UserConstants
+import no.nav.syfo.testhelper.createDialogmotekandidatEndring
+import no.nav.syfo.testhelper.dropData
+import no.nav.syfo.testhelper.generator.*
+import no.nav.syfo.unntak.api.domain.toUnntak
+import no.nav.syfo.unntak.database.createUnntak
+import org.amshove.kluent.internal.assertFailsWith
+import org.amshove.kluent.shouldBeEqualTo
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.describe
+import java.time.LocalDate
+import java.time.OffsetDateTime
+
+object IdenthendelseServiceSpek : Spek({
+
+    describe(IdenthendelseServiceSpek::class.java.simpleName) {
+
+        with(TestApplicationEngine()) {
+            start()
+
+            val externalMockEnvironment = ExternalMockEnvironment.instance
+            val database = externalMockEnvironment.database
+            val azureAdClient = AzureAdClient(
+                azureEnvironment = externalMockEnvironment.environment.azure
+            )
+            val pdlClient = PdlClient(
+                azureAdClient = azureAdClient,
+                pdlEnvironment = externalMockEnvironment.environment.clients.pdl,
+            )
+
+            val identhendelseService = IdenthendelseService(
+                database = database,
+                pdlClient = pdlClient,
+            )
+
+            beforeEachTest {
+                database.dropData()
+            }
+
+            describe("Happy path") {
+                it("Skal oppdatere tabeller når person har fått ny ident") {
+                    val kafkaIdenthendelseDTO = generateKafkaIdenthendelseDTO(hasOldPersonident = true)
+                    val newIdent = kafkaIdenthendelseDTO.getActivePersonident()!!
+                    val oldIdenter = kafkaIdenthendelseDTO.getInactivePersonidenter()
+
+                    populateDatabase(oldIdenter.first(), database)
+
+                    val oldIdentOccurrences = database.getIdentOccurenceCount(oldIdenter)
+                    oldIdentOccurrences shouldBeEqualTo 4
+
+                    runBlocking {
+                        identhendelseService.handleIdenthendelse(kafkaIdenthendelseDTO)
+                    }
+
+                    val newIdentOccurrences = database.getIdentOccurenceCount(listOf(newIdent))
+                    newIdentOccurrences shouldBeEqualTo 4
+                }
+
+                it("Skal oppdatere tabeller der gammel ident forekommer når person har fått ny ident") {
```

Comment:
Hva er egentlig forskjellen på disse to testene, annet den gamle identen finnes i hhv. fire og to rader?

Reply:
Det var litt for å teste at det ikke oppsto noen "rariteter" når vi bare gjør updates på noen av tabellene (og ikke alle). Men synes du det ikke er vits? Det funker, i hvertfall 🤷🏼‍♂️